paddle/fluid/platform/enforce.h

```diff
@@ -191,6 +195,35 @@ struct BinaryCompareMessageConverter<false> {
 };
 }  // namespace details
 
+template <typename T>
+inline std::string ReplaceComplexTypeStr(std::string str,
+                                         const std::string& type_name) {
+  auto demangle_type_str = demangle(typeid(T).name());
+  size_t start_pos = 0;
+  while ((start_pos = str.find(demangle_type_str, start_pos)) !=
+         std::string::npos) {
+    str.replace(start_pos, demangle_type_str.length(), type_name);
+    start_pos += type_name.length();
+  }
+  return str;
+}
+
+#define __REPLACE_COMPLEX_TYPE_STR__(__TYPENAME, __STR)                       \
+  do {                                                                        \
+    __STR = paddle::platform::ReplaceComplexTypeStr<__TYPENAME>(__STR,        \
+                                                                #__TYPENAME); \
+  } while (0)
+
+inline std::string SimplifyDemangleStr(std::string str) {
+  // the older is important, you have to put complex types in front
```

Comment:
Not important.
Why should put complex types in front?

Reply:
for example, if we replace `Attribute` firstly, then we can't found `AttributeMap`, because the part of `AttributeMap` had been replaced, but actually, we can replace whole `AttributeMap` firstly.